Pull Request (squashed commits)

Textarea-LotC opened the pull request:
I couldn't figure out how to squash them inside of my master branch, so here's another branch lol

Commit:
Attempting to fix two bugs

Bug#1 = Word-wrapped item name loses formatting
Bug#2 = Right bracket gains formatting from the item name
Thing I dislike: "Showing held item" comes after the item is shown

Converting legacy text into text components

Commit:
/countdown

Commit:
/emote

Commit:
Reducing /roll broadcast radius to 20

Commit:
Added countdown and emote to plugin.yml

Archemedes reviewed (changes requested):
Consider the following:
- for stylistics please use camelCase for all variables. so PlayersNearby would be playersNearby
- there is a specific terminal stream operation called .collect( ). It is preferred to do this if your terminal operation is to add all entries found to a list, as you are doing.
- its not needed to have an async runnable just to send a message. It doesnt make it faster if thats what you think. Just run within the actual  method at the calling point
- clearing the PlayersNearby object is not garbage collecting (this works very different in java). in fact doing this will cause all of your runnables AFTER this to not function because the list would have been emptied before the runnables run (since they are delayed)
- the @Flag annotation has a "permission" attribute specifically to restrict it to certain people. Doing it this way would make sure the documentation reflects people's ability to use the flag (dont show flags peiople cant use). Please use @Flag(permission="...") instead of checking it manually inside the method.

Tofuus commented on src/main/java/co/lotc/lever/cmd/Countdown.java:
```diff
@@ -0,0 +1,52 @@
+    @Flag(name="r",description="Countdown broadcast radius",type=double.class)
+    public void invoke(Player source, @Arg("Timer") @Default("5") @Range(min=5,max=60) int timer) {
+        if (COOLDOWN_LIST.contains(source.getUniqueId())) {
+            msg(RED + "Command on cooldown.");
```
Use the validate() method that all the other commands use

Archemedes replied:
validate() is admittedly bad practice when it comes to java readability :100: so its not mandatory to use it. But it'd be good to see what validate() is because it saves some boilerplate if you like it. I like it myself.

Tofuus replied:
It's not about readability here but consistency. But don't Most projects have some form of @NonNull or whatever

Tofuus commented on src/main/java/co/lotc/lever/cmd/ShowItem.java:
```diff
@@ -18,9 +18,11 @@ public void invoke(Persona source, Player target) {
-		
-		b.send(target);
+				.append('[').color(WHITE)
+				.append(ItemUtil.getDisplayName(is)).hoverItem(is)
```
Legacy system needs to be converted to text components here

Archemedes commented on src/main/java/co/lotc/lever/cmd/Countdown.java:
```diff
@@ -0,0 +1,54 @@
+        Run.as(plugin).delayed(delayInTicks, () -> playersNearby.stream().filter(Objects::nonNull).forEach(player -> player.sendMessage(BOLD + "" + DARK_GREEN + "PvP Begins!")));
+
+        delayInTicks += 20L;
+        Run.as(plugin).delayed(delayInTicks, playersNearby::clear);
```
This really isn't necessary

Archemedes commented:
Looks good to me now. I think we can merge this alongside Chat Update

Archemedes commented:
Testing github webhook

Textarea-LotC commented:
Changed countdown range to be 3-10 (the min & max of the current countdown plugin)
Moved `validate()` to be right under `invoke()` in Countdown.java
Removed `Run.as(plugin).delayed(delayInTicks, playersNearby::clear);` from Countdown.java

Squashed some commits & force-pushed :yum:

Tofuus commented on src/main/java/co/lotc/lever/cmd/Countdown.java:
```diff
@@ -0,0 +1,52 @@
+import java.util.List;
+
+public class Countdown extends BaseCommand {
+    public static final String PEX_COUNTDOWN_RADIUS = "lever.countdown.radius";
```
Permission not pex

Archemedes replied:
thats my fault i use PEX as shorthand everywhere in lEver

Tofuus commented on src/main/java/co/lotc/lever/cmd/Emote.java:
```diff
@@ -0,0 +1,26 @@
+
+public class Emote extends BaseCommand {
+
+    @Flag(name="r",description="Emote broadcast radius",type=double.class)
```
I'd prefer the radius be a default not a flag

Archemedes replied:
This is a good place to use flags I'd rather people learn to start using them becuase theyre much more informative in the long run.

Tofuus reviewed (changes requested):
Looks fine to me. Just address the minor stuff.

Textarea-LotC committed:
Merge branch 'master' into test

Commit:
/fly list

Getting a list of online players who are server-side flying

Is this really all you have to do to make an argument for a command? I feel like I'm missing something...
